File: src/main/java/fi/metatavu/oioi/cm/rest/translate/MediaTranslator.java
package fi.metatavu.oioi.cm.rest.translate;

/**
 * Translator for Media REST entity 
 * 
 * @author Antti Leppä
 */
public class MediaTranslator extends AbstractTranslator<fi.metatavu.oioi.cm.persistence.model.Media, fi.metatavu.oioi.cm.model.Media> {

  @Override
  public fi.metatavu.oioi.cm.model.Media translate(fi.metatavu.oioi.cm.persistence.model.Media entity) {
    if (entity == null) {
      return null;
    }

    fi.metatavu.oioi.cm.model.Media result = new fi.metatavu.oioi.cm.model.Media();    
    result.setId(entity.getId());
    result.setContentType(entity.getContentType());
    result.setType(entity.getType());
    result.setUrl(entity.getUrl());
    result.setCreatorId(entity.getCreatorId());
    result.setLastModifierId(entity.getLastModifierId());
    result.setCreatedAt(entity.getCreatedAt());
    result.setModifiedAt(entity.getModifiedAt());
    return result;
  }
  
}

File: src/main/kotlin/fi/metatavu/oioi/cm/rest/translate/MediaTranslator.kt
package fi.metatavu.oioi.cm.rest.translate

import fi.metatavu.oioi.cm.persistence.model.Media
import javax.enterprise.context.ApplicationScoped

/**
 * Translator for Media REST entity
 *
 * @author Antti Leppä
 */
@ApplicationScoped
class MediaTranslator : AbstractTranslator<Media?, fi.metatavu.oioi.cm.model.Media?>() {
    override fun translate(entity: Media?): fi.metatavu.oioi.cm.model.Media? {
        if (entity == null) {
            return null
        }

        return fi.metatavu.oioi.cm.model.Media(
            id = entity.id,
            contentType = entity.contentType!!,
            type = entity.type!!,
            url = entity.url!!,
            creatorId = entity.creatorId,
            lastModifierId = entity.lastModifierId,
            createdAt = entity.createdAt,
            modifiedAt = entity.modifiedAt
        )
    }
}